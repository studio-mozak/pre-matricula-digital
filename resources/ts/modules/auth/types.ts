export interface Auth {
  level: number;
  name: string;
  schools: string[];
}
