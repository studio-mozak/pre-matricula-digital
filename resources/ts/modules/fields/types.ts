import { Nullable } from '@/types';

export type SetValues = (fields: Record<string, unknown>) => void;

export interface FieldPageOption {
  id?: string;
  name: Nullable<string>;
  weight: number;
}

export interface FieldPage {
  id: Nullable<string>;
  name: Nullable<string>;
  description: Nullable<string>;
  group: Nullable<'RESPONSIBLE' | 'STUDENT'>;
  type: Nullable<string>;
  internal: boolean;
  options: FieldPageOption[];
  deleteOptions?: string[];
  required?: boolean;
}

export interface ErrorResponse {
  message: string;
}
