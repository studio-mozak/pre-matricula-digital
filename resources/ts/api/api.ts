import axios, { AxiosInstance } from 'axios';

const token = window.config ? window.config.token : '';

export const http: AxiosInstance = axios.create({
  baseURL: String(import.meta.env.VITE_APP_API_URL || ''),
  headers: {
    Authorization: `Token ${token}`,
  },
});

axios.interceptors.response.use(
  (res) => {
    /**
     * If the response from the server has the "errors" object or the object "data" is empty,
     * here it is forced to go to the catch block automatically.
     */
    if (
      (res.data.errors || Object.keys(res.data).length === 0) &&
      !(res.data instanceof Blob)
    ) {
      return Promise.reject({
        response: {
          data: res.data,
        },
      });
    }

    return res;
  },
  (err) => Promise.reject(err)
);

export const rest = <T>(
  type: 'get' | 'post' | 'put' | 'delete',
  url: string,
  params?: object
): Promise<T> => http[type](url, params);

export const graphql = <T>(data: object, config?: object): Promise<T> =>
  axios.post('/graphql', data, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
    ...config,
  });
