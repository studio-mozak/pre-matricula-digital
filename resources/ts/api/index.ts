export * from './api';
export * as City from './services/graphql/city';
export * as Resource from './services/graphql/resource';
