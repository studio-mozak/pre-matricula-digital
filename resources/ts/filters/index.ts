import { Address, GoogleMapsMarker } from '@/types';
import {
  getFormattedDate,
  getFormattedDateTime,
  getFormattedTime,
} from '@/datetime';
import {
  joinAddress,
  joinAddressLine1,
  joinAddressLine2,
  joinAddressLine3,
  markerAddress,
} from '@/util';

export interface Filters {
  formatDateTime: (date: string) => string;
  formatDate: (date: string) => string;
  formatTime: (date: string) => string;
  joinAddress: (address: Address) => string;
  joinAddressLine1: (address: Address) => string;
  joinAddressLine2: (address: Address) => string;
  joinAddressLine3: (address: Address) => string;
  markerAddress: () => GoogleMapsMarker;
}

const filters: Filters = {
  formatDateTime(date: string) {
    return getFormattedDateTime(date);
  },
  formatDate(date: string) {
    return getFormattedDate(date);
  },
  formatTime(date: string) {
    return getFormattedTime(date);
  },
  joinAddress,
  joinAddressLine1,
  joinAddressLine2,
  joinAddressLine3,
  markerAddress,
};

export default filters;
