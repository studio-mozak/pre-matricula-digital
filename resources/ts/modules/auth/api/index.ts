/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as AuthRest from '@/modules/auth/api/services/rest/auth';
