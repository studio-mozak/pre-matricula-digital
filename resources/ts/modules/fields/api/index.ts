/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Field from './services/graphql/field';
