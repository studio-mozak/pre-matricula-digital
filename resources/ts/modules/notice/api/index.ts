/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Notice from './services/graphql/notice';
