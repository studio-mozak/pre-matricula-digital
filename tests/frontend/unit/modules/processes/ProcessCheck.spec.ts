import '@/plugin/sortby';
import { VueWrapper, flushPromises, mount } from '@vue/test-utils';
import { ProcessCheck as IProcessCheck } from '@/modules/processes/types';
import { Process as ProcessApi } from '@/modules/processes/api';
import ProcessCheck from '@/modules/processes/components/ProcessCheck.vue';
import VTooltip from 'v-tooltip';
import { createTestingPinia } from '@pinia/testing';
import filters from '@/filters';
import { expect, vi } from 'vitest';
import { analytics } from '@/packages';
import { AnalyticsInstance } from 'analytics';

interface ProcessCheckProps {
  id?: string;
  newProcess: string;
}

vi.mock('@/modules/processes/api');
vi.mock('vue-router/dist/vue-router.mjs', () => ({
  useRouter: () => ({
    push: spyPush,
  }),
}));
vi.mock('@/packages');

const spyPush = vi.fn();

const processCheckMock: IProcessCheck = {
  id: '1',
  name: 'Ed. Infantil (Nível III, Pré-Escola) Ens. Fundamental (1° ao 9° ano)',
  fields: [
    {
      field: {
        group: 'RESPONSIBLE',
        id: '1',
        internal: 'responsible_name',
        name: 'Nome do(a) responsável',
        options: [],
        type: 'TEXT',
      },
      id: '184',
      order: 1,
      required: true,
      weight: 0,
    },
  ],
  grades: [
    {
      endBirth: '2017-03-31',
      id: '21',
      name: 'Nível III',
      startBirth: '2016-03-31',
    },
  ],
  periods: [
    {
      id: '1',
      name: 'Matutino',
    },
  ],
  schoolYear: { year: 2021 },
  schools: [
    {
      id: '000000',
      latitude: 0,
      longitude: 0,
      name: 'ESCOLA MUNICIPAL DE ENSINO FUNDAMENTAL',
    },
  ],
  stages: [
    {
      endAt: '2026-03-07 23:59:00',
      id: '17',
      name: 'Test',
      startAt: '2021-02-17 14:00:00',
      status: 'OPEN',
      type: 'REGISTRATION',
    },
  ],
  vacancies: [
    {
      available: 54,
      grade: '6',
      period: '2',
      school: '98370',
      total: 81,
    },
  ],
};

const onMounted = async (
  props: ProcessCheckProps,
  processCheck: {
    success: boolean;
    data: IProcessCheck;
  }
) => {
  if (processCheck.success) {
    vi.mocked(ProcessApi.show).mockResolvedValueOnce(processCheck.data);
  } else {
    vi.mocked(ProcessApi.show).mockRejectedValueOnce(processCheck.data);
  }

  vi.mocked(analytics).mockImplementation(
    () =>
      ({
        page: vi.fn(),
      } as unknown as AnalyticsInstance)
  );

  const wrapper = mount(ProcessCheck, {
    props,
    global: {
      plugins: [createTestingPinia()],
      stubs: ['x-field'],
      provide: {
        $filters: filters,
      },
      directives: {
        tooltip: VTooltip,
      },
    },
  });

  await wrapper.vm.$nextTick();

  await flushPromises();

  return wrapper;
};

describe('ProcessCheck', () => {
  let wrapper: VueWrapper<InstanceType<typeof ProcessCheck>>;

  beforeEach(async () => {
    wrapper = await onMounted(
      {
        newProcess: 'false',
        id: '1',
      },
      {
        success: true,
        data: processCheckMock,
      }
    );

    await wrapper.vm.$nextTick();
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  test('the component exists', () => {
    expect(wrapper).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('the initial variables must be loaded with the correct data', () => {
    expect(wrapper.vm.student).toMatchObject({
      grade: null,
      period: null,
      school: null,
      useSecondSchool: false,
      secondSchool: null,
      secondPeriod: null,
      match: null,
    });
    expect(wrapper.vm.responsible).toMatchObject({
      relationType: null,
      address: {
        postalCode: '',
        address: '',
        number: '',
        complement: '',
        neighborhood: '',
        lat: 0,
        lng: 0,
        city: '',
        cityIbgeCode: 0,
        stateAbbreviation: '',
      },
      useSecondAddress: false,
      secondAddress: {
        postalCode: '',
        address: '',
        number: '',
        complement: '',
        neighborhood: '',
        lat: 0,
        lng: 0,
        city: '',
        cityIbgeCode: 0,
        stateAbbreviation: '',
      },
      responsible_name: null,
    });
    expect(wrapper.vm.fields.responsible.length).toBe(1);
    expect(wrapper.vm.fields.student.length).toBe(0);
  });

  describe('Error on initial requisition', () => {
    beforeEach(async () => {
      wrapper = await onMounted(
        {
          newProcess: 'false',
          id: '1',
        },
        {
          success: false,
          data: {} as IProcessCheck,
        }
      );

      await wrapper.vm.$nextTick();
    });

    afterEach(() => {
      vi.clearAllMocks();
    });

    test('the initial variables must be loaded with the correct data', () => {
      expect(wrapper.vm.student).toMatchObject({});
      expect(wrapper.vm.responsible).toMatchObject({});
      expect(wrapper.vm.fields.responsible.length).toBe(0);
      expect(wrapper.vm.fields.student.length).toBe(0);
    });
  });

  test('if I click on "go back" button, the user should be redirected to previous page', async () => {
    const button = wrapper.find('[data-test="btn-back"]');

    await button.trigger('click');

    expect(spyPush).toBeCalled();
    expect(spyPush).toBeCalledTimes(1);
    expect(spyPush).toBeCalledWith({
      name: 'process.periods',
      params: {
        id: '1',
        newProcess: 'false',
      },
    });
  });

  describe('submit', () => {
    test('if I click on "submit" button, the user should be redirected to next page', async () => {
      const button = wrapper.find('[data-test="btn-submit"]');

      await button.trigger('click');

      expect(spyPush).toBeCalled();
      expect(spyPush).toBeCalledTimes(1);
      expect(spyPush).toBeCalledWith({
        name: 'processes',
      });
    });
  });
});
