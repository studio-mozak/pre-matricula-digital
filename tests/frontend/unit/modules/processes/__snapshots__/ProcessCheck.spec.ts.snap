// Vitest Snapshot v1

exports[`ProcessCheck > should match snapshot 1`] = `
<main
  class="container"
  style="max-width: 740px;"
>
  <h2
    class="font-muli-20-primary"
  >
    Crie o processo de pré-matrícula
  </h2>
  <p>
     Revise os dados do processo a ser criado e prossiga para finalizá-lo. Você pode pré-visualizar os formulários criados para esse processo e editá-los se necessário. 
    <strong>
       Lembre-se que depois de criado, um processo não poderá ser editado. 
    </strong>
  </p>
  
  <div
    class="d-flex justify-content-between mt-5"
  >
    <h4>
      Ed. Infantil (Nível III, Pré-Escola) Ens. Fundamental (1° ao 9° ano) (2021)
    </h4>
  </div>
  <div
    class="row"
  >
    
    <div
      class="col-12 mb-3"
    >
      <div
        class="x-card x-card--bordered x-card--hover"
      >
        <div
          class="x-card__section x-card__section--vert pb-0"
        >
          <h5
            class="text-h5"
          >
            Matrícula
          </h5>
        </div>
        <div
          class="x-card__section x-card__section--vert pt-0"
        >
          <div
            class="mt-2 d-flex align-items-center text-size-15"
          >
            <i
              class="fa fa-calendar text-primary mr-2"
            />
            <span
              class="text-gray-600"
            >
              17/02/2021 às 14:00 à 07/03/2026 às 23:59
            </span>
          </div>
          <div
            class="mt-3"
          >
            
            <small
              class="badge-green badge mr-2"
            >
              Matutino
            </small>
            
          </div>
        </div>
      </div>
    </div>
    
  </div>
  <div
    class="mt-5 mb-4"
  >
    <h4>
      Formulários
    </h4>
  </div>
  <div
    class="row"
  >
    <div
      class="col-12"
    >
      <div
        class="d-flex justify-content-between align-items-center border-bottom pb-4 mb-4"
      >
        <p>
          Dados do(a) responsável
        </p>
        <div>
          <button
            class="x-btn x-btn-item non-selectable no-outline x-btn--outline x-btn--rectangle text-primary border-primary x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase"
            role="button"
            tabindex="0"
            type="submit"
          >
            <span
              class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
            >
              <span
                class="block"
              >
                Editar
              </span>
            </span>
            <transition-stub
              appear="false"
              css="true"
              name="x-transition--fade"
              persisted="false"
            >
              <!---->
            </transition-stub>
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        
        <div
          class="col-12 col-md-6 mb-3"
        >
          <label
            class="toggle-checkbox"
          >
            <input
              disabled=""
              type="checkbox"
              value="responsible_name"
            />
            <div
              class="toggle-required toggle-text"
            >
              Nome do(a) responsável
            </div>
          </label>
        </div>
        
        <div
          class="col-12 mb-3"
        >
          <label
            class="toggle-checkbox"
          >
            <input
              disabled=""
              type="checkbox"
            />
            <div
              class="toggle-text toggle-required"
            >
              Endereço
            </div>
          </label>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-12"
        >
          <button
            class="x-btn x-btn-item non-selectable no-outline x-btn--standard x-btn--rectangle bg-primary text-white x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase w-100"
            role="button"
            style="font-size: 20px;"
            tabindex="0"
            type="button"
          >
            <span
              class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
            >
              <span
                class="block"
              >
                Pré-visualizar formulário
              </span>
            </span>
            <transition-stub
              appear="false"
              css="true"
              name="x-transition--fade"
              persisted="false"
            >
              <!---->
            </transition-stub>
          </button>
        </div>
        <!--v-if-->
      </div>
    </div>
    <div
      class="col-12 mt-5"
    >
      <div
        class="d-flex justify-content-between align-items-center border-bottom pb-4 mb-4"
      >
        <p>
          Dados do(a) aluno(a)
        </p>
        <div>
          <button
            class="x-btn x-btn-item non-selectable no-outline x-btn--outline x-btn--rectangle text-primary border-primary x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase"
            role="button"
            tabindex="0"
            type="submit"
          >
            <span
              class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
            >
              <span
                class="block"
              >
                Editar
              </span>
            </span>
            <transition-stub
              appear="false"
              css="true"
              name="x-transition--fade"
              persisted="false"
            >
              <!---->
            </transition-stub>
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        
        
      </div>
      <div
        class="row"
      >
        <div
          class="col-12"
        >
          <button
            class="x-btn x-btn-item non-selectable no-outline x-btn--standard x-btn--rectangle bg-primary text-white x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase w-100"
            role="button"
            style="font-size: 20px;"
            tabindex="0"
            type="button"
          >
            <span
              class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
            >
              <span
                class="block"
              >
                Pré-visualizar formulário
              </span>
            </span>
            <transition-stub
              appear="false"
              css="true"
              name="x-transition--fade"
              persisted="false"
            >
              <!---->
            </transition-stub>
          </button>
        </div>
        <!--v-if-->
      </div>
    </div>
  </div>
  <div
    class="d-flex justify-content-between align-items-end mt-5 mb-3"
  >
    <h4
      class="m-0"
    >
      Critérios
    </h4>
    <div>
      <button
        class="x-btn x-btn-item non-selectable no-outline x-btn--outline x-btn--rectangle text-primary border-primary x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase"
        role="button"
        tabindex="0"
        type="submit"
      >
        <span
          class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
        >
          <span
            class="block"
          >
            Editar
          </span>
        </span>
        <transition-stub
          appear="false"
          css="true"
          name="x-transition--fade"
          persisted="false"
        >
          <!---->
        </transition-stub>
      </button>
    </div>
  </div>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-12 text-muted"
    >
       Nenhum critério de prioridade foi definido. 
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-12"
    >
      
      
    </div>
  </div>
  <div
    class="row mt-5"
  >
    <div
      class="offset-md-3 col-md-3"
    >
      <button
        class="x-btn x-btn-item non-selectable no-outline x-btn--outline x-btn--rectangle text-gray-500 border-gray-500 x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase w-100 h-100 flex-row justify-content-center"
        data-test="btn-back"
        role="button"
        style="font-size: 20px;"
        tabindex="0"
        type="submit"
      >
        <span
          class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
        >
          <span
            class="block"
          >
            Voltar
          </span>
        </span>
        <transition-stub
          appear="false"
          css="true"
          name="x-transition--fade"
          persisted="false"
        >
          <!---->
        </transition-stub>
      </button>
    </div>
    <div
      class="col-md-3 mt-3 mt-md-0"
    >
      <button
        class="x-btn x-btn-item non-selectable no-outline x-btn--standard x-btn--rectangle bg-primary text-white x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase w-100 h-100 flex-row justify-content-center"
        data-test="btn-submit"
        role="button"
        style="font-size: 20px;"
        tabindex="0"
        type="submit"
      >
        <span
          class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
        >
          <span
            class="block"
          >
            Prosseguir
          </span>
        </span>
        <transition-stub
          appear="false"
          css="true"
          name="x-transition--fade"
          persisted="false"
        >
          <!---->
        </transition-stub>
      </button>
    </div>
  </div>
  
</main>
`;
